### PR TITLE
Added path info to RazorPage exceptions

### PR DESCRIPTION
**Pull request by ajaybhargavb**
Issue - #1919 
The exception referenced by the issue has already been fixed. 
I've updated few other exceptions in `RazorPage` and `RazorView`.
@Eilon @pranavkm

### CODE REVIEW

File: src/Microsoft.AspNet.Mvc.Razor/RazorView.cs [hunk 1]
```diff
@@ -179,7 +179,7 @@ private bool EnableInstrumentation
                     // the body content. Throwing this exception wouldn't return a 500 (since content has already been
                     // written), but a diagnostic component should be able to capture it.
 
-                    var message = Resources.FormatLayoutCannotBeRendered("FlushAsync");
+                    var message = Resources.FormatLayoutCannotBeRendered(Path, "FlushAsync");
```
Comment (dougbu) [inline src/Microsoft.AspNet.Mvc.Razor/RazorView.cs]: `nameof(RazorPage.FlushAsync)` while you're here

File: src/Microsoft.AspNet.Mvc.Razor/Resources.resx [hunk 2]
```diff
@@ -148,7 +148,7 @@
     <value>There is no active writing scope to end.</value>
   </data>
   <data name="RazorPage_YouCannotFlushWhileInAWritingScope" xml:space="preserve">
-    <value>You cannot flush while inside a writing scope.</value>
+    <value>You cannot flush while inside a writing scope. (Path: '{0}')</value>
```
Comment (dougbu) [inline src/Microsoft.AspNet.Mvc.Razor/Resources.resx]: leave out the colon; not needed when you're already quoting the path.  same for `SectionAlreadyRendered`

File: src/Microsoft.AspNet.Mvc.Razor/Resources.resx [hunk 3]
```diff
@@ -133,7 +133,7 @@
     <value>The layout view '{0}' could not be located. The following locations were searched:{1}</value>
   </data>
   <data name="LayoutCannotBeRendered" xml:space="preserve">
-    <value>A layout page cannot be rendered after '{0}' has been invoked.</value>
+    <value>A layout page cannot be rendered at '{0}' after '{1}' has been invoked.</value>
```
Comment (dougbu) [inline src/Microsoft.AspNet.Mvc.Razor/Resources.resx]: - updated sentence does not scan.  suggest `Layout page '{1}' cannot be rendered after '{0}' has been invoked.`
- I reversed the parameters to match `SectionAlreadyRendered` but that's a nit

File: src/Microsoft.AspNet.Mvc.Razor/Resources.resx (hunk 2 above)
Comment (dougbu) [inline src/Microsoft.AspNet.Mvc.Razor/Resources.resx]: also suggest `'{0}' invocation in '{1}' is invalid. That method must not be used inside a writing scope.`

File: src/Microsoft.AspNet.Mvc.Razor/Resources.resx [hunk 4]
```diff
@@ -163,7 +163,7 @@
     <value>Section '{0}' is already defined.</value>
   </data>
   <data name="SectionAlreadyRendered" xml:space="preserve">
-    <value>The section named '{0}' has already been rendered.</value>
+    <value>The section named '{0}' has already been rendered. (Path: '{1}')</value>
```
Comment (dougbu) [inline src/Microsoft.AspNet.Mvc.Razor/Resources.resx]: suggest `'{0}' invocation in '{2}' is invalid. Section '{1} has already been rendered.` where caller passes `nameof(RenderSectionAsync)` for the 0th argument.

**Comment by dougbu** [comment]
:watch:

**Comment by pranavkm** [comment]
Couple more places where we could print out the path:

* https://github.com/aspnet/Mvc/blob/render-body-fix/src/Microsoft.AspNet.Mvc.Razor/RazorPage.cs#L611-L612
> '{method name}' invocation in '{Path}' is invalid.

* https://github.com/aspnet/Mvc/blob/render-body-fix/src/Microsoft.AspNet.Mvc.Razor/RazorPage.cs#L820

* https://github.com/aspnet/Mvc/blob/render-body-fix/src/Microsoft.AspNet.Mvc.Razor/RazorPage.cs#L724
> Section '{section}' is not defined in '{Path}'.

File: src/Microsoft.AspNet.Mvc.Razor/Resources.resx (hunk 3 above)
Comment (Eilon) [inline src/Microsoft.AspNet.Mvc.Razor/Resources.resx]: :+1:

File: src/Microsoft.AspNet.Mvc.Razor/Resources.resx (hunk 2 above)
Comment (Eilon) [inline src/Microsoft.AspNet.Mvc.Razor/Resources.resx]: `invocation` sounds weird and so does `that method`... Though the original string said `you cannot flush` which while true, is irrelevant: people can never flush anything while inside a writing scope. It's more like `The XYZ operation cannot be performed in ABC mode in 'PATH'.` or something, no?

Comment (Eilon) [inline src/Microsoft.AspNet.Mvc.Razor/Resources.resx]: (And change the string name while we're at it - the word `you` is so weird in an error message.)

File: src/Microsoft.AspNet.Mvc.Razor/RazorView.cs (hunk 1 above)
Comment (ajaybhargavb) [inline src/Microsoft.AspNet.Mvc.Razor/RazorView.cs]: `FlushAsync` is not part of `IRazorPage`. So can't access it from here.

Comment (dougbu) [inline src/Microsoft.AspNet.Mvc.Razor/RazorView.cs]: the issue is the `RazorPage` property hiding the class.  but `nameof(Razor.RazorPage.FlushAsync)` works.

**Comment by ajaybhargavb** [comment]
Updated.

File: src/Microsoft.AspNet.Mvc.Razor/RazorPage.cs [hunk 5]
```diff
@@ -748,14 +748,14 @@ public Task<HtmlString> RenderSectionAsync([NotNull] string name)
             // change.
             if (_writerScopes.Count > 0)
             {
-                throw new InvalidOperationException(Resources.RazorPage_YouCannotFlushWhileInAWritingScope);
+                throw new InvalidOperationException(Resources.FormatRazorPage_CannotFlushWhileInAWritingScope(nameof(FlushAsync), Path));
```
Comment (dougbu) [inline src/Microsoft.AspNet.Mvc.Razor/RazorPage.cs]: nit: long

File: test/Microsoft.AspNet.Mvc.Razor.Test/RazorPageTest.cs [hunk 6]
```diff
@@ -236,7 +239,8 @@ public class RazorPageTest
             await page.ExecuteAsync();
 
             // Assert
-            Assert.Equal("RenderSection can only be called from a layout page.",
+            Assert.Equal("RenderSection invocation in '/Views/TestPath/Test.cshtml' is invalid. " +
+                "RenderSection can only be called from a layout page.",
                          ex.Message);
```
Comment (dougbu) [inline test/Microsoft.AspNet.Mvc.Razor.Test/RazorPageTest.cs]: nit: odd indentation; left-shift this line

**Comment by dougbu** [comment]
:shipit:

File: src/Microsoft.AspNet.Mvc.Razor/Resources.resx (hunk 2 above)
Comment (pranavkm) [inline src/Microsoft.AspNet.Mvc.Razor/Resources.resx]: Does this apply to the other messages that were changed in this PR?

**Comment by Eilon** [comment]
:shipit: from me too

**Comment by pranavkm** [comment]
:shipit:

**Pull request merged**